jazz_environments/components/crud/getList.js

```diff
@@ -0,0 +1,93 @@
+// =========================================================================
+// Copyright © 2017 T-Mobile USA, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// =========================================================================
+
+/**
+    Get List of Service-Catalogs
+    @module: getList.js
+    @description: CRUD functions for service catalog
+    @author:
+    @version: 1.0
+ **/
+
+const utils = require("../utils.js")(); //Import the utils module.
+const logger = require("../logger.js"); //Import the logging module.
+
+module.exports = (query, onComplete) => {
+    // initialize dynamodb
+    var dynamodb = utils.initDynamodb();
+
+    var filter = "";
+    var attributeValues = {};
+
+    var insertAndString = " AND ";
+
+    var scanparams = {
+        TableName: global.env_tableName,
+        ReturnConsumedCapacity: "TOTAL",
+        Limit: "500"
+    };
+
+    if (query) {
+        // var keys_list = ['service', 'domain', 'region', 'type', 'runtime', 'created_by','timestamp'];
+        var keys_list = global.config.service_environment_filter_params;
+
+        // Generate filter string
+        keys_list.forEach(function(key) {
+            var key_name = utils.getEnvironmentDatabaseKeyName(key);
+
+            if (query[key]) {
+                filter = filter + key_name + " = :" + key_name + insertAndString;
+                attributeValues[":" + key_name] = {
+                    S: query[key]
+                };
+            }
+        });
+    }
+
+    filter = filter.substring(0, filter.length - insertAndString.length); // remove the " AND " at the end
+
+    if (filter !== "") {
+        scanparams.FilterExpression = filter;
+        scanparams.ExpressionAttributeValues = attributeValues;
+    }
+
+    var items_formatted = [];
+    var scanExecute = function (onComplete) {
```

Comment:
Use Query instead of scan as in internal (new version)

Reply:
Using query

Reply:
As discussed not integrating query